prov/gni/include/gnix_freelist.h

```diff
@@ -121,8 +123,8 @@ static inline int _gnix_fl_alloc(struct dlist_entry **e, struct gnix_freelist *f
 
     if (fl->ts)
         fastlock_acquire(&fl->lock);
-
-    if (dlist_empty(&fl->freelist)) {
+    
+    if (dlist_empty(&fl->freelist) && fl->refill_size) {
```

Comment:
Does this fall out properly if fl->refill_size == 0?  In other words, what happens down below when you try to access fl->freelist.next?